pkg/util/wait/wait.go

```diff
@@ -153,15 +156,29 @@ type Backoff struct {
 // errors terminate immediately.
 func ExponentialBackoff(backoff Backoff, condition ConditionFunc) error {
 	duration := backoff.Duration
-	for i := 0; i < backoff.Steps; i++ {
-		if i != 0 {
-			adjusted := duration
-			if backoff.Jitter > 0.0 {
-				adjusted = Jitter(duration, backoff.Jitter)
+
+	// always run once
+	if ok, err := condition(); err != nil || ok {
+		return err
+	}
+
+	for i := 0; i < (backoff.Steps - 1); i++ {
+		adjusted := duration
+		// respect the max duration if set
+		if adjusted > 0 {
```

Comment:
this should be `if backoff.MaxDuration > 0 {`, right? If this is actually a bug, there should have been a test to catch it.